packages/server/src/utils/KeyvLRU.ts

```diff
@@ -0,0 +1,85 @@
+import LRUCache from 'lru-cache';
+import Keyv, { Store, type Options } from 'keyv';
+
+// LRUCache wrapper to implement the Keyv `Store` interface.
+export class LRUStore<T> implements Store<T> {
+  private cache: LRUCache<string, T>;
+
+  constructor(lruCacheOpts: LRUCache.Options<string, T>) {
+    this.cache = new LRUCache(lruCacheOpts);
+  }
+
+  set(key: string, value: T, ttl?: number) {
+    const result = this.cache.set(key, value, { ttl });
+    return result;
+  }
+
+  get(key: string) {
+    return this.cache.get(key);
+  }
+
+  delete(key: string) {
+    return this.cache.delete(key);
+  }
+
+  clear() {
+    this.cache.clear();
+  }
+
+  keys() {
+    const keys = [];
+    for (const key of this.cache.keys()) {
+      keys.push(key);
+    }
+    return keys;
+  }
+}
+
+export class KeyvLRU<T> extends Keyv<T, { store: LRUStore<T> }> {
+  constructor(opts?: Options<T> & { store: LRUStore<T> }) {
+    super({
+      namespace: 'apollo',
+      store: new LRUStore<T>({
+        // FIXME: @glasser this was previously Infinity but no longer allowed,
```

Comment:
To make sure I follow, KeyvLRU is used in:

1) DocumentStore which overrides `store`
2) The default `cache` which does not override `store` and that's the thing that used to be infinite (I guess, haven't double checked this)
3) Various places in testing (where we shouldn't care much about the max)

I guess the choice is between
a) Making the default (used in 2 and 3) be bounded
b) Keeping the default unbounded

Has the current behavior (where cache features by default have an unbounded cache and you're recommended to replace it with a specified cache in production) been a problem? I'd lean towards just making the default not use LRUStore at all... isn't the default if you don't pass a `store` to keyv that it uses a Map or something?

BTW are we exporting KeyvLRU? Seems like it might be a useful thing that people might want to use in their own server, eg to specify a `cache: KeyvLRU( some specific max stuff )`.  Although maybe that should be in `apollo-utils`...

Reply:
(or like, we just publish a package like https://www.npmjs.com/package/keyv-lru or https://github.com/sindresorhus/quick-lru that works with lru-cache.  we did evaluate these two and decided that the caches they wrap aren't as good as lru-cache, right?)

Reply:
(ah yes, i think neither of them do size calculation.  which i guess we think is important :) but publishing a simple keyv-lru-cache outside of AS could be helpful?  like just move the current `LRUStore` to be in a new keyv-lru-cache package? (Which we could publish via `apollo-utils` if that's easy...)

then do we even need KeyvLru at all or do we just have people do `new Keyv({ store: new keyvLruCache({ ... }) })`

am i missing something?

Reply:
I didn't explore those in favor of continuing to use `lru-cache` and changing less if possible. I'm pretty open to trying one of these out and dropping our need for a wrapper at all.

Ok, a couple commits later:
* The `cache` option now defaults to a vanilla `Keyv()` which uses a plain `Map` for a store.
* The `KeyvLRU` is now gone, it really had effectively just become `Keyv` with `store: new LRUStore()`
* Some defaults now live in `LRUStore()` (size calculator and max size)

07c190a
4599c16